src/vuex/modules/index/types.ts

```diff
@@ -1,11 +1,110 @@
+export class Index {
+  private _name: string
+  loading = false
+  collections?: Collection[]
+
+  constructor(name: string) {
+    this._name = name
+  }
+
+  get name() {
+    return this._name
+  }
+
+  get CollectionsCount(): number | undefined {
+    return this.collections?.length
+  }
+
+  public initCollections(collections: Collection[]) {
```

Comment:
About putting logic in the types.

It is very appealing to do it this way, since we all like to properly separate the code. But beware: since this code is no longer in a mutation, it can now be called outside a mutation, which will break the Vuex paradigm (which, let's not forget it, is the only thing that makes a global object like the store viable). In the current implementation, the logic is always called by the mutations (I think, I hope) but... Are we sure we will always remember to do so?
According to Murphy's Law, the answer is no.

Reply:
What can we do to prevent Murphy's Law or, at least, mitigate its prophecy? Ideas...
* Giving up the OOP-style stuff and just putting the logic in the mutations (after all, Vuex/Redux/Flux paradigms are all about functional programming, but I know this is not the kind of things you want to read at this point).
* Writing JSDoc comments on _all_ state-mutating methods that would _try to_ warn the developer about not calling these methods outside a mutation.
* Adding some weird `fn.caller` logic that would trigger an error if the method is not called by a mutation (dunno how to do this).

😬